### ISSUE TITLE
Fix data discrepancies in federal data

### ISSUE THREAD

**Issue by amitness** [CodeforNepal/nepalmap_federal#19]
During the conversion of federal data to the format needed for Wazimap in #11, there was variation in the geography.sql mapping and the data scraped at [federal-data](https://github.com/Code4Nepal/data/tree/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold). 

Creating this separate issue to track that.

**Comment by cliftonmcintosh** [comment]
These are all of the locations that appear in the geography data but do not appear in the population data. There are two types of localities that are problematic:

1. municipalities and one district
2. parks and reserves

The parks and reserves could posslby be filled in in the population data with zero populations. The missing district and muncipalities cannot be filled in that way, though.


These are the locations that have no population data and are not parks or reserves.
```
wazimap_np=> select g.geo_level, g.geo_code, g.name, pg.name as parent_name, g.parent_level, g.parent_code from wazimap_geography g join wazimap_geography pg on (g.parent_level = pg.geo_level and g.parent_code = pg.geo_code) where  g.parent_level = pg.geo_level and g.parent_code = pg.geo_code and g.name not like '%Park%' and g.name not like '%Reserve%' and g.geo_code not in (select geo_code from population p where g.geo_level = p.geo_level) order by g.parent_level, parent_name, g.name; 
 geo_level | geo_code |                name                 |   parent_name    | parent_level | parent_code 
-----------+----------+-------------------------------------+------------------+--------------+-------------
 local     | 57003    | Janki                               | Banke            | district     | 07
 local     | 30013    | Tripura Sundari                     | Dhading          | district     | 17
 local     | 17001    | Aaurahi                             | Dhanusa          | district     | 19
 local     | 61002    | Bheri                               | Jajarkot         | district     | 27
 local     | 71004    | Dhangadhi                           | Kailali          | district     | 30
 local     | 25004    | Lalitpur                            | Lalitpur         | district     | 38
 local     | 48001    | Bardaghat                           | Nawalparasi West | district     | 76
 local     | 48002    | Palhi Nandan                        | Nawalparasi West | district     | 76
 local     | 48003    | Pratappur                           | Nawalparasi West | district     | 76
 local     | 48004    | Ramgram                             | Nawalparasi West | district     | 76
 local     | 48005    | Sarawal                             | Nawalparasi West | district     | 76
 local     | 48006    | Sunwal                              | Nawalparasi West | district     | 76
 local     | 48007    | Susta                               | Nawalparasi West | district     | 76
 local     | 28012    | Tarkeshwar                          | Nuwakot          | district     | 48
 local     | 49099    | Lumbini Sanskritik Development Area | Rupandehi        | district     | 60
 local     | 49015    | Sudhdhodhan                         | Rupandehi        | district     | 60
 local     | 6010     | Itahari                             | Sunsari          | district     | 69
 local     | 39008    | Kaligandagi                         | Syangja          | district     | 71
 district  | 76       | Nawalparasi West                    | Province No. 5   | province     | 5
(19 rows)

```

These are the parks and reserves that have no population data and might logically be filled in with zeros.
```
wazimap_np=> select g.geo_level, g.geo_code, g.name, pg.name as parent_name, g.parent_level, g.parent_code from wazimap_geography g join wazimap_geography pg on (g.parent_level = pg.geo_level and g.parent_code = pg.geo_code) where  g.parent_level = pg.geo_level and g.parent_code = pg.geo_code and g.name like '%Park%' or g.name like '%Reserve%' and g.geo_code not in (select geo_code from population p where g.geo_level = p.geo_level) order by g.parent_level, parent_name, g.name; 
 geo_level | geo_code |                   name                   |   parent_name    | parent_level | parent_code 
-----------+----------+------------------------------------------+------------------+--------------+-------------
 local     | 69099    | Khaptad National Park                    | Achham           | district     | 01
 local     | 45099    | Dhorpatan Hunting Reserve                | Baglung          | district     | 03
 local     | 68099    | Khaptad National Park                    | Bajhang          | district     | 05
 local     | 67099    | Khaptad National Park                    | Bajura           | district     | 06
 local     | 33099    | Parsa Wildlife Reserve                   | Bara             | district     | 08
 local     | 58099    | Bardiya National Park                    | Bardiya          | district     | 09
 local     | 35099    | Chitawan National Park                   | Chitwan          | district     | 12
 local     | 70099    | Khaptad National Park                    | Doti             | district     | 22
 local     | 72099    | Shuklaphanta National Park               | Kanchanpur       | district     | 32
 local     | 31099    | Chitawan National Park                   | Makwanpur        | district     | 41
 local     | 31088    | Parsa Wildlife Reserve                   | Makwanpur        | district     | 41
 local     | 43099    | Dhorpatan Hunting Reserve                | Myagdi           | district     | 46
 local     | 76099    | Chitawan National Park                   | Nawalparasi East | district     | 47
 local     | 28099    | Langtang National Park                   | Nuwakot          | district     | 48
 local     | 28088    | Shivapuri Watershed And Wildlife Reserve | Nuwakot          | district     | 48
 local     | 34099    | Chitwan National Park                    | Parsa            | district     | 53
 local     | 77099    | Dhorpatan Hunting Reserve                | Rukum East       | district     | 59
 local     | 15099    | Koshi Tappu Wildlife Reserve             | Saptari          | district     | 63
 local     | 6099     | Koshi Tappu Wildlife Reserve             | Sunsari          | district     | 69
 local     | 14099    | Koshi Tappu Wildlife Reserve             | Udayapur         | district     | 75
(20 rows)


```

**Comment by cliftonmcintosh** [comment]
I believe these are some of the mappings for missing localities:
-  "Janki" in Banke district is "Janaki" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%205/Banke/Janaki%20Gaunpalika.csv
- "Tripura Sundari" in Dhadhing district is "Tripurasundari Gaunpalika" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%203/Dhading/Tripurasundari%20Gaunpalika.csv
- "Aaurahi" in Dhanusa district is "Aurahi Gaunpalika" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%202/Dhanusa/Aurahi%20Gaunpalika.csv
- "Bheri" in "Jajarkot" is "Bheri Malika Municipality" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%206/Jajarkot/Bheri%20Malika%20Municipality.csv
- "Dhangadhi" in "Kailali" is "Dhangadhi Sub-Metropolitian City" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%207/Kailali/Dhangadhi%20Sub-Metropolitian%20City.csv
- "Lalitpur" in "Lalitpur" is "Lalitpur Metropolitian City" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%203/Lalitpur/Lalitpur%20Metropolitian%20City.csv
- "Tarkeshwar" in "Nuwakot" is "Tarakeshwor Gaunpalika" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%203/Nuwakot/Tarakeshwor%20Gaunpalika.csv
- "Sudhdhodhan" in "Rupandehi" is "Suddhodhan Gaunpalika" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%205/Rupandehi/Suddhodhan%20Gaunpalika.csv
- "Itahari" in "Sunsari" is "Itahari Sub-Metropolitian City" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%201/Sunsari/Itahari%20Sub-Metropolitian%20City.csv
- "Kaligandagi" in "Syangja" is "Kaligandaki Gaunpalika" in the federal population data: https://github.com/Code4Nepal/data/blob/master/Federal%20Data/753%20Local%20Unit%20Population%20and%20HouseHold/PRADESH%204/Syangja/Kaligandaki%20Gaunpalika.csv

I'm not yet sure how we manage the data for Nawalparasi West. In discussing the population data via email, @nikeshbalami said: 
> the CBS data still consists of the old name. Here is the new districts name for your reference, I will also update this name once the old PR is merged.
> Nawalparasi: Nawalpur District (Province 5) and Parasi District (Province 4)
> Rukum:Eastern Rukum (Province 5) and Western Rukum (Province 6)

I'm also not sure what "Lumbini Sanskritik Development Area" is or where it's data might be. There is already a  "Lumbini Sanskritik" separate from the "Lumbini Sanskritik Development Area".

**Comment by cliftonmcintosh** [comment]
I have created #23 to track what to do about parks and reserves.

**Comment by cliftonmcintosh** [comment]
@amitness 

Do you have scripts available that can be shared (and used) for updating the data for the discrepancies? We still have some localities which are problematic, but I was considering trying to tackle the ones I identified last weekend. If you have scripts, would you be willing to commit them to the maptools project?

**Comment by amitness** [comment]
@cliftonmcintosh I have the script but it was done in Jupyter notebook so the code is vey rough. I've structered it into separate groups and added comments. It's present in my fork here: https://github.com/amitness/maptools/tree/dev/federal/analysis 

To run, 
`jupyter notebook federal-data.ipynb`

Assumes that `pandas` and `jupyter` are installed.
`pip install pandas jupyter`

**Comment by nikeshbalami** [comment]
The issue was created because of the missing datasets of the Parasi districts from the Federal Data repo, I just submitted the PR with updated datasets.

**Regarding the district**
Nawalparasi: [Nawalpur, Province 4 (Nawalparasi East)] [Parasi, Province 5 (Nawalparasi West)] - We need to use the data of Parasi District at Nawalparasi West.

**Regarding the missing local bodies data**
The data of entire Parasi District was missing/replace by Kapilbastu data. I just scrapped and has submitted the PR of those datasets. The remaining are already pointed by Cliff at the issue.
- “Bardaghat” as “Bardaghat Municipality
- “Palhi Nandan” as “Palhinandan Gaunpalika”
- “Pratappur” as “Pratapapur Gaunpalika”
- “Ramgram” as “Ramgram Municipality”
- “Sarawal” as “Sarawal Gaunpalika”
- “Sunwal” as “Sunawal Municipality”
- “Susta” as “Susta Gaunpalika”

**Regarding the parks and reservation centers**
We can use “0” for all of them and "Lumbini Sanskritik" and ”Lumbini Sanskritik Development Area" are same. We can use “0” for both.

**Comment by cliftonmcintosh** [comment]
With @nikeshbalami's data updates and his guidance provided in the comment directly above, we should now have the ability to account for the missing data. @amitness has some tooling for processing data in a Jupyter notebook. See this comment: https://github.com/Code4Nepal/nepalmap_federal/issues/19#issuecomment-437935324.

@ravinepal if one of the volunteers wants to try to process the missing data, they are welcome to do so. This would include:

- including data for the missing localities that have now all been identified in the data set.
- including zero values for all national parks, wildlife reserves and development districts.
- updating the aggregated totals for all levels above the local level so that they include the new figures. For example, if a missing locality is added to Banke district, then the totals for that district needs to be updated to included the locality's figures. The totals for Province 5 would need to be updated to include the locality's figures, and the totals for Nepal would need to be updated to include the locality's figures.

**Comment by ravinepal** [comment]
@suyogdahal - would you be interested in helping with this?

**Comment by suyogdahal** [comment]
@ravinepal  - Sure sir, I'll have a look at it !

**Comment by cliftonmcintosh** [comment]
@suyogdahal 

I have broken this into separate issues, one for each province to make it easier for someone to submit data for just one province. If you have not yet started and would like to take this on province-by-province, then please use one of those issues for your work. My reasoning for this is that it will allow volunteers to contribute in smaller batches and may allow multiple volunteers to work at the same time.

If, however, you have already started doing all of the provinces, please let me know, and I will close the issue.

**Comment by cliftonmcintosh** [comment]
I have created the following issues to make it easier to break up the work into manageable units:

https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/32
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/33
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/34
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/35
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/36
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/37
https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/31https://github.com/Code4Nepal/nepalmap_federal/issues/38

**Comment by cliftonmcintosh** [comment]
@suyogdahal 

I wanted to let you know that I have added data for national parks, hunting and wildlife reserves and development areas. #40 is the PR for that. That PR fulfills the issue #38.

How is it going with other data? Do you have any questions? Is there anything that @amitness or @crackjack or I can provide guidance about?

**Comment by cliftonmcintosh** [comment]
Fixed with
#40 
#41 
#42 
#45 

Thank you, @pratimakandel !
